homework/index.js

```diff
@@ -27,21 +31,136 @@
         elem.setAttribute(key, value);
       }
     });
+
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  // fetch contributor
+  async function fetchContributorsAndRender(contributorUrl) {
+    try {
+      const contributors = await fetchJSON(contributorUrl);
```

Comment:
I liked how you have split logic of your program in separate functions and each function doing one thing only. 🙂